dfastmi/io/RiverParameterData.py

```diff
@@ -219,183 +226,175 @@ def collect_values1(
             branch one float.
         """
         self.is_initialized()
-        try:
-            g_val = self._config["General"][key]
-        except:
-            g_val = ""
-
-        all_values = []
-        for ib in range(len(self._branches)):
-            branch = self._branches[ib]
+        val = self.__read_value(key, branch, reach)
+        if val == "" and default is not None:
+            ival = default
+        else:
             try:
-                b_val = self._config[branch][key]
+                vals = tuple(int(x) for x in val.split())
             except:
-                b_val = g_val
+                vals = ()
+            if len(vals) != 1:
+                raise Exception(
+                    'Reading {} for reach {} on branch {} returns "{}". Expecting {} values.'.format(
+                        key, reach.name, branch.name, val, 1
+                    )
+                )
+            ival = vals[0]
+        return ival
+    
+    def read_key_float(
+            self,
+            key: str,
+            branch: Branch,
+            reach: Reach,
+            default: Optional[float] = None,
+        ) -> float:
+        """
+        This routines collects entries of type float.
 
-            values_per_branch = []
-            for i in range(self._nreaches[ib]):
-                stri = str(i + 1)
-                try:
-                    val = self._config[branch][key + stri]
-                except:
-                    val = b_val
-                if val == "" and default is not None:
-                    fval = default
-                else:
-                    try:
-                        vals = tuple(float(x) for x in val.split())
-                    except:
-                        vals = ()
-                    if len(vals) != 1:
-                        raise Exception(
-                            'Reading {} for reach {} on {} returns "{}". Expecting {} values.'.format(
-                                key, i+1, self._branches[ib], val, 1
-                            )
-                        )
-                    fval = vals[0]
-                values_per_branch.append(fval)
+        Arguments
+        ---------
+        key : str
+            The name of the parameter for which the values are to be retrieved.
+        branch : Branch
+            The branch where we want to read the parameter from
 
-            all_values.append(values_per_branch)
 
-        return all_values
+        Raises
+        ------
+        Exception
+            If the number of values read from the file doesn't match 1.
+
+        Returns
+        -------
+        data : List[List[float]]
+            A list of lists. Each list contains per reach within the corresponding
+            branch one float.
+        """
+        self.is_initialized()
+        val = self.__read_value(key, branch, reach)
+        if val == "" and default is not None:
+            fval = default
+        else:
+            try:
```

Comment:
feels like read_key_bool & read_key_int, is there a way to deduplicate this?

Reply:
leave it up to [DFAST-112](https://issuetracker.deltares.nl/browse/DFAST-112)

Reply:
Discussed, we will leave these comments for a later refactoring.
This will be tackled in https://issuetracker.deltares.nl/browse/DFAST-112